Produce consume tutorial in scala (#459)
confluentinc/kafka-tutorials: _includes/tutorials/produce-consume-lang/scala/markup/test/2_0-make-test-sources-dir.adoc

Let's create a directory for the tests to live in:

+++++
<pre class="snippet"><code class="bash">{%
    include_raw tutorials/produce-consume-lang/scala/code/tutorial-steps/test/make-test-sources-dir.sh
%}</code></pre>
+++++

Before we go further we need to discuss *what* and *how* to test in these applications.

> *What* to test? We don't want to test Kafka clients them self's. Instead, we'd like to test functions with
> business behavior, even if they have side effects due to the call to Kafka producers / consumers.

Reviewer: ```suggestion
*What* to test? We don't want to test Kafka clients themselves. Instead, we'd like to test functions with
business behavior, even if they have side effects due to the call to Kafka producers / consumers.
```

> *How* to test? On the opposite of most tutorials on this site, we are not testing streaming applications here.
> Which means we can not extract a streaming topology and test it separately.
> We have to spawn a real Kafka broker to test our functions. And to do this we will use the
> https://https://www.testcontainers.org/modules/kafka/[testcontainers] Kafka module.

Reviewer: ```suggestion
*How* to test? We are going to use a real Kafka broker to test our functions, and to do this we will use Test Containers and the https://https://www.testcontainers.org/modules/kafka/[testcontainers Kafka module].
```

Author: I really thought the comparison with the Kafka Streams Test Utils was important for someone already familiar with other tutorials.

Reviewer: If you feel strongly that it's good for the student, please feel free to leave it!
